### ISSUE TITLE
Controlling visibility of volume node children in subject hierarchy

### ISSUE THREAD

**Issue by gregsharp** [Slicer/Slicer#4961]
The visibility toggle button (ie. the eyeball button) in the subject hierarchy can be used to show or hide the children of the hierarchy node.  However, visibility of children volume nodes is not affected in the 2D slice views.

## Steps to reproduce

See discourse post. 
https://discourse.slicer.org/t/hiding-patient-in-hierarchy/11850

## Expected behavior

Clicking the visibility button of a parent should allow user to control all children nodes in a uniform manner.

## Environment
- Slicer version: 4.11.0-2020-03-26 r28898 / f9eb7e9
- Operating system: Linux

**Comment by cpinter** [comment]
Volume visibility is very different from the visibility of any other data type in Slicer. Only two volumes can be shown in each slice view, and they need to be assigned roles (background, foreground).

* What do you do when there are three volumes in the folder you want to show?
* What do you do if there are two volumes shown and you click on the eye button of another volume?
* What do you do if different slice views show different volumes?

We have had different implementations giving different answers to these questions, and none of those proved to be acceptable for everyone, mainly because there are so many use cases and possible scenarios.
So we decided to handle volume visibility differently, and thus assigned a different show/hide icon to indicate this.

If we reach a consensus about how to approach this, then I think we are happy to change the behavior. But unfortunately after years of trying out different implementations, and having multiple breakouts at project weeks about this, we still haven't found a better solution.

**Comment by lassoan** [comment]
Lifting the limitation of showing only two volumes could be a solution, but of course it would need a lot of changes.

**Comment by pieper** [comment]
Yes, at the lowest levels of the code you could have an arbitrary list of volumes to display, but you'd need to have control over the compositing order.  Early on we decided that two layers (BG/FG) was a good compromise in terms of UI complexity so that's baked into a lot of the API.

**Comment by lassoan** [comment]
Yes, even overlaying two volumes makes it very hard to interpret them, so showing more than that usually would not make sense (more layers could be useful for image mosaics, multispectral images, etc. but specific alternative solutions could be developed for these).

**Maybe what users would really want is to show additional views!** When user clicks on show volume while another volume is already displayed, we could switch to a view layout with multiple view groups (such as three-over-three, four-over-four).

**Comment by cpinter** [comment]
To me switching layout would be very unexpected. Changing layout on show/hide is unprecedented in Slicer, so probably others would find it surprising too.

How about showing a list of options when the user clicks volume visibility? Whatever default action we chose is unexpected for some, so popping up four or five options would probably be welcome.

However, neither changing layouts or popping up options would solve the original question, which is showing volumes when showing the parent folder. Unless in that case we show a more complex set of options (there may be more volumes), but I don't have a good feeling about showing a complex popup about something implicit (click show volume vs click show study).

**Comment by fbordignon** [comment]
Thanks for linking this thread. I know it is a challenge from the development perspective, but it is really confusing for the user to click on the eye of a folder and only certain itens show/hide like I've reported on #6413. Perhaps try to check if all display nodes of itens in the folder are group showable or hidable before taking any action, because clicking a folder and hiding a 3D volume render and not hiding the 2D slices in the **same** view is really confusing in my opinion.

Another option would be to force hide the volumes inside a folder on the 2D slices. This should indicate that the folder is hidden and the user must open the folder eye to view the 2D Slices. When opening the eye, there is no need to show all the volumes/itens inside the folder, only the ones with the eye opened. Slicer does not let one open the eye of multiple scalar volumes at the same time anyways.
